### PR TITLE
profiles: initial commits for riscv profile for prefix

### PR DESCRIPTION
**Pull request by wiredhikari**

### CODE REVIEW

**Comment by amadio** [comment]
There's a submodule added here `net-p2p/prefix`, can you remove?

File: profiles/prefix/linux/riscv/eapi [hunk 1]
```diff
@@ -0,0 +1 @@
+5
```
Comment (amadio) [inline profiles/prefix/linux/riscv/eapi:1]: This should probably be 8.

File: profiles/prefix/linux/riscv/make.defaults [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+ARCH="riscv"
+CHOST="riscv64-pc-linux-gnu"
+# The base profile sets ACCEPT_KEYWORDS=riscv and we don't have that in prefix.
```
Comment (amadio) [inline profiles/prefix/linux/riscv/make.defaults:6]: This comment is false. The `riscv` arch is unstable only in Gentoo as far as I know.

File: profiles/prefix/linux/riscv/make.defaults [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+ACCEPT_KEYWORDS="~riscv"
+
+# We don't have lib64 in prefix so, remove it here.
+SYMLINK_LIB=""
+LIBDIR_riscv="lib"
```
Comment (amadio) [inline profiles/prefix/linux/riscv/make.defaults:11]: I think this needs to be either `LIBDIR_lp64="lib"` or `LIBDIR_lp64d="lib"`, see `default/linux/riscv/20.0/rv64gc/lp64d/musl/make.defaults` for example.

File: profiles/prefix/linux/riscv/make.defaults [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Copyright 1999-2022 Gentoo Authors
```
Comment (dlan17) [inline profiles/prefix/linux/riscv/make.defaults:1]: it's fine to start from 2022 here

File: profiles/profiles.desc [hunk 5]
```diff
@@ -310,6 +310,7 @@ amd64-linux		prefix/linux/amd64				exp
 arm-linux		prefix/linux/arm				exp
 ppc64-linux		prefix/linux/ppc64				exp
 ppc64-linux		prefix/linux/ppc64le				exp
+riscv-linux		prefix/linux/riscv				exp
```
Comment (dlan17) [inline profiles/profiles.desc:313]: ditto here

File: profiles/prefix/linux/riscv/packages [hunk 6]
```diff
@@ -0,0 +1,2 @@
+# Copyright 1999-2022 Gentoo Authors
```
Comment (dlan17) [inline profiles/prefix/linux/riscv/packages:1]: ditto

File: profiles/prefix/linux/riscv/parent [hunk 7]
```diff
@@ -0,0 +1,2 @@
+../../../default/linux/riscv/20.0/rv64gc
```
[the remark this answers is not in the document]
Comment (amadio) [reply profiles/prefix/linux/riscv/parent:1]: Yeah, reading https://gcc.gnu.org/onlinedocs/gcc/RISC-V-Options.html makes me think that we should go for lp64d only, and add support for more ABIs later on an as needed basis.

File: profiles/prefix/linux/riscv/parent [hunk 8]
```diff
@@ -0,0 +1,2 @@
+../../../default/linux/riscv/20.0/rv64gc/ip64d
```
Comment (amadio) [inline profiles/prefix/linux/riscv/parent:1]: This should be `lp64d`, not `ip64d`, right?

**Comment by amadio** [comment]
Indeed, thanks for the reminder. I'm adding it now.

**Pull request closed without merging**